hudi-spark-datasource/hudi-spark2/src/main/java/org/apache/hudi/internal/HoodieDataSourceInternalWriter.java

```diff
@@ -61,13 +60,15 @@ public HoodieDataSourceInternalWriter(String instantTime, HoodieWriteConfig writ
     this.structType = structType;
     this.populateMetaFields = populateMetaFields;
     this.arePartitionRecordsSorted = arePartitionRecordsSorted;
-    this.extraMetadataMap = DataSourceUtils.getExtraMetadata(dataSourceOptions.asMap());
+    this.sparkSession = sparkSession;
+    Map<String, String> extraMetadataMap = DataSourceUtils.getExtraMetadata(dataSourceOptions.asMap());
     this.dataSourceInternalWriterHelper = new DataSourceInternalWriterHelper(instantTime, writeConfig, structType,
         sparkSession, configuration, extraMetadataMap);
   }
 
   @Override
   public DataWriterFactory<InternalRow> createWriterFactory() {
+    sparkSession.sparkContext().setJobGroup(this.getClass().getSimpleName(), "Writing data to files using bulk_insert", true);
```

Comment:
Let's avoid setting misleading annotations -- this method is just creating the factory but not actually writing any data until later point in time.